### PR TITLE
Custom sizing

### PR DESCRIPTION
**Pull request by omawhite**
adds docs for kuberentes component sizing

### CODE REVIEW

File: reference/halyard/component-sizing.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
+---
+layout: single
+title:  "Component Sizing"
+sidebar:
+  nav: reference
+redirect_from: /docs/component-sizing
```
Comment (ttomsu) [inline reference/halyard/component-sizing.md]: No need to include a `redirect_from`. This was only needed for old docs that transferred from the readme.io site.

**Comment by ttomsu** [comment]
Thanks for the docs update! I'm performing some profiling this week across VMs and containers, so if you don't mind, can we sit on this for a week or so? I may have some stuff to add to make it more general than just distributed Kubernetes.

**Comment by edwinavalos** [comment]
We can definitely wait on it! Thanks for letting us know.

**Comment by edwinavalos** [comment]
@omawhite looks like the link i made to reference another page is broken, so we'll change that tomorrow morning, the details of the broken link are in the failed CI test.

**Comment by omawhite** [comment]
@edwinavalos i think i fixed it, the tests passed locally on my latest commit.

File: reference/halyard/component-sizing.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        cpu: 250m
+        memory: 512Mi
+      requests:
+        cpu: 100m
+        memory: 128Mi
+```
+
+Limits and requests follow the Kubernetes conventions [documented here](https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/).
+
+### Updating JAVA_OPTS
```
Comment (rosskevin) [inline reference/halyard/component-sizing.md]: Is this section necessary when potentially using `JAVA_DOCKER_MEMORY_OPTS='-XX:+UnlockExperimentalVMOptions -XX:+UseCGroupMemoryLimitForHeap -XX:MaxRAMFraction=1 -XshowSettings:vm'` described by @lwander @greenkiwi [here](https://github.com/spinnaker/halyard/issues/683#issuecomment-329791497)?

Comment (edwinavalos) [reply reference/halyard/component-sizing.md]: @rosskevin yes, that setting only works on a newer version of Java than all the microservices are using at the moment.

background here: https://github.com/spinnaker/halyard/pull/724#issuecomment-335577794

Comment (ttomsu) [reply reference/halyard/component-sizing.md]: Related: https://github.com/spinnaker/spinnaker/issues/2058

**Comment by edwinavalos** [comment]
@ttomsu did we want to hold off on this still? I've been pointing people to this PR instead of the website a few times now.

**Comment by ttomsu** [comment]
I've also found this PR incredibly helpful :-) 

The first question many people are going to have is "This capability is great - but what values should I use for each service?" And that obviously depends on multiple factors of your environment, so I don't think we'll be able to provide that kind of guidance, other than "X, Y, and Z are the largest services, A, B, and C are medium, the rest are small"

Maybe we should include something like that in this doc.

**Comment by omawhite** [comment]
@edwinavalos @ttomsu I took a crack at modifying the recommendations, also added a little intro sentence under the kuberentes header on the page. take a look.

**Comment by ttomsu** [comment]
@dorbin Can you take a look?

File: reference/halyard/component-sizing.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+layout: single
+title:  "Component Sizing"
+sidebar:
+  nav: reference
+---
+
+{% include toc %}
+
+
+Within the halconfig custom sizing can be specified for each Spinnaker component. Custom sizing must be manually added to the halconfig, there are no CLI commands or flags to enable these settings. This feature is currently only supported for distributed deployments using Kubernetes (`hal config deploy edit --type distributed --account-name my-k8s-cluster`).
```
Comment (dorbin) [inline reference/halyard/component-sizing.md]: For "halconfig," I recommend saying either "the Halyard configuration" or "~/.hal/config"

File: reference/halyard/component-sizing.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+## Kubernetes
+The following kubernetes settings can be tweaked within custom sizing.
```
Comment (dorbin) [inline reference/halyard/component-sizing.md]: s/kubernetes/Kubernetes

File: reference/halyard/component-sizing.md (hunk 3 above)
Comment (dorbin) [reply reference/halyard/component-sizing.md]: ...in all cases

File: reference/halyard/component-sizing.md (hunk 4 above)
Comment (dorbin) [reply reference/halyard/component-sizing.md]: ...in all cases

**Review by dorbin — APPROVED** [review]
Nice, Omar! Thanks for the contribution!

**Comment by markus-silpala-tgt** [comment]
We'll do that. Thanks, guys.

**Pull request merged**